### PR TITLE
Make DNS/DoH recommendations

### LINKED ISSUE #9: Recommend disabling DoH when using mullvad
There's no privacy benefits to use DoH when using Mullvad.
If a user has DoH set, we should recommend them to disable it.

Also, see #11

### PR DESCRIPTION
**Pull request by ruihildt**
Closes  #9  #11  #126

And replace icons  #97

### CODE REVIEW

**Comment by hankolsen** [comment]
> @hankolsen I added an icon to the eslint ignore file, becaus eI couldn't figure out how to fix the reported issues.
> 
> Can you have a look?

Hmm, I removed the entry in .eslintignore and ran `npm run lint` without any issues.
What errors are you getting?

File: src/components/Icons/FeCheckCircle.vue [hunk 1]
```diff
@@ -1,10 +1,15 @@
 <template>
-  <svg width="1em" height="1em" viewBox="0 0 24 24">
-    <path
-      d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10s-4.477 10-10 10zM8 10l-2 2l5 5l7-7l-2-2l-5 5l-3-3z"
-      fill="currentColor"
-      fillRule="nonzero"
-    ></path>
+  <svg
+    width="24"
```
Comment (hankolsen) [inline src/components/Icons/FeCheckCircle.vue:3]: By using `1em` as width/height you can control the size of the icon by the font size of the parent.

**Comment by ruihildt** [comment]
> Hmm, I removed the entry in .eslintignore and ran `npm run lint` without any issues.
> What errors are you getting?
I had some errors regarding some spacing eslint rules. In turn, this would then stop me from committing anything.

Works for me too now, probably a bug in my editor which got fixed by restarting my computer.

**Comment by ruihildt** [comment]
> Am I supposed to see all these at the same time?

These will show in the home view among the recommendations when they are not properly configured.
When they are listed with the green checkmark, it'because they are not relevant.

We could probably hide them instead, do you feel it makes more sense?

**Comment by hankolsen** [comment]
> > Am I supposed to see all these at the same time?
> 
> These will show in the home view among the recommendations when they are not properly configured. When they are listed with the green checkmark, it'because they are not relevant.
> 
> We could probably hide them instead, do you feel it makes more sense?

I got confused by the two first ones referring to different states, I was connected to Mullvad VPN at the time but still got saw the text about not being connected.

**Comment by ruihildt** [comment]
Ok then this is probably a bug.

You should only see one DNS/DoH recommendation at a time, based on your current settings.

**Comment by hankolsen** [comment]
> Ok then this is probably a bug.
> 
> You should only see one DNS/DoH recommendation at a time, based on your current settings.

I'm connected to Mullvad, I have not changed any dns settings in Firefox.
All of these are shown with a green check mark:
* Disable Mullvad DoH (encrypted DNS)
* Enable Mullvad DoH (encrypted DNS)
* DoH leaks have been detected
* DNS leaks have been detected

**Comment by ruihildt** [comment]
> I'm connected to Mullvad, I have not changed any dns settings in Firefox.
> All of these are shown with a green check mark:

The green checkmark indicates the recommendation is followed: if there's something to do, the checkmark is replaced by a red warning exclamation point.

I think part of the confusion comes from having opposite actions (enable AND disable) shown as green checkmarked at the same time

**Comment by ruihildt** [comment]
Maybe these shouldn't be displayed as recommendations at all, as they are actually warning about leaks.

**Comment by hankolsen** [comment]
This looks a bit sqashed:
image

**Comment by hankolsen** [comment]
Clicking on the individual "Learn more" links on the first page opens up the Privacey Recommendations and correctly scrolls the user to the relevent section.
However, clicking on `Show all` at the bottom of the first page opens up the Privacy Recommendations, but not at the top of the page.

File: src/components/ConnectionNotifications.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          @click="closePopup"
+        >
+          Read more
+        </Button>
+      </div>
+    </div>
+  </n-card>
+</template>
+
+<style>
```
Comment (hankolsen) [inline src/components/ConnectionNotifications.vue:42]: These styles doesn't seem to be used here anymore?

**Review by hankolsen — CHANGES_REQUESTED** [review]

**Comment by ruihildt** [comment]
> However, clicking on `Show all` at the bottom of the first page opens up the Privacy Recommendations, but not at the top of the page.

Fixed.

File: src/components/ConnectionNotifications.vue (hunk 2 above)
Comment (ruihildt) [reply src/components/ConnectionNotifications.vue]: Removed.

**Review by hankolsen — APPROVED** [review]
The only thing is the `Use HTTPS-only mode` section which looks a bit messy, otherwise it looks fine!

**Comment by ruihildt** [comment]
> The only thing is the `Use HTTPS-only mode` section which looks a bit messy, otherwise it looks fine!

Just pushed a fix:
image

**Pull request merged**